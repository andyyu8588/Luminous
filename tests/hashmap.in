// For compiler/VM testing purpose

import HashMap

class Object {
    public id;
    public message;

    public constructor(id, message) {
        this.id = id;
        this.message = message;
    }
}

function simpleHash(num) {
    return (num * 31) % 11;
}

map = HashMap(simpleHash);

for(i from 0 to 100 by 1) {
    map.insert(i, Object(i, "This object has id " + i));
}

// Replacement test:
for(i from 0 to 100 by 1) {
    map.insert(i, Object(i, "This object has ID " + i));
}


for(i from 0 to 100 by 1) {
    object = map.find(i);
    print(object.message);
}

for(i from 0 to 50 by 1) {
    map.delete(i);
    print(map.find(i));
}

for(i from 99 to 49 by -1) {
    map.delete(i);
    print(map.find(i));
}
