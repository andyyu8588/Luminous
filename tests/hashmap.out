This object has id 0
This object has id 1
This object has id 2
This object has id 3
This object has id 4
This object has id 5
This object has id 6
This object has id 7
This object has id 8
This object has id 9
This object has id 10
This object has id 11
This object has id 12
This object has id 13
This object has id 14
This object has id 15
This object has id 16
This object has id 17
This object has id 18
This object has id 19
This object has id 20
This object has id 21
This object has id 22
This object has id 23
This object has id 24
This object has id 25
This object has id 26
This object has id 27
This object has id 28
This object has id 29
This object has id 30
This object has id 31
This object has id 32
This object has id 33
This object has id 34
This object has id 35
This object has id 36
This object has id 37
This object has id 38
This object has id 39
This object has id 40
This object has id 41
This object has id 42
This object has id 43
This object has id 44
This object has id 45
This object has id 46
This object has id 47
This object has id 48
This object has id 49
This object has id 50
This object has id 51
This object has id 52
This object has id 53
This object has id 54
This object has id 55
This object has id 56
This object has id 57
This object has id 58
This object has id 59
This object has id 60
This object has id 61
This object has id 62
This object has id 63
This object has id 64
This object has id 65
This object has id 66
This object has id 67
This object has id 68
This object has id 69
This object has id 70
This object has id 71
This object has id 72
This object has id 73
This object has id 74
This object has id 75
This object has id 76
This object has id 77
This object has id 78
This object has id 79
This object has id 80
This object has id 81
This object has id 82
This object has id 83
This object has id 84
This object has id 85
This object has id 86
This object has id 87
This object has id 88
This object has id 89
This object has id 90
This object has id 91
This object has id 92
This object has id 93
This object has id 94
This object has id 95
This object has id 96
This object has id 97
This object has id 98
This object has id 99
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
